crates/wasm-pkg-loader/src/config.rs

```diff
@@ -73,13 +73,10 @@ impl ClientConfig {
     pub fn set_oci_registry_config(
         &mut self,
         registry: impl Into<String>,
-        client_config: Option<OciClientConfig>,
+        client_config: OciClientConfig,
         credentials: Option<BasicCredentials>,
     ) -> Result<&mut Self, Error> {
-        if client_config
-            .as_ref()
-            .is_some_and(|cfg| cfg.platform_resolver.is_some())
-        {
+        if client_config.platform_resolver.is_some() {
             Error::InvalidConfig(anyhow::anyhow!(
```

Comment:
I was a bit puzzled by this line. It looks like it constructs the Error and then drops it rather than returning it. But when I tried changing it to return the Error, the CI failed with this error message. So I've left it along but I feel like I'm missing something...!

Reply:
That is definitely broken!